File: Example/WZRootNavigationController_Example_ObjC/ViewControllers/WZCustomAnimatedTransitionListCollectionViewController.m
//
//  WZCustomAnimatedTransitionListCollectionViewController.m
//  WZRootNavigationController
//
//  Created by 吴哲 on 2018/4/25.
//  Copyright © 2018年 CocoaPods. All rights reserved.
//

#import "WZCustomAnimatedTransitionListCollectionViewController.h"
#import "WZAnimatedCollectionViewCell.h"
#import "WZRootNavigationController_Example_ObjC-Swift.h"

@interface WZCustomAnimatedTransitionListCollectionViewController ()

@end

@implementation WZCustomAnimatedTransitionListCollectionViewController

static NSString * const reuseIdentifier = @"Cell";

- (Class)wz_animatedTransitionPluginClass
{
    return WZCustomAnimatedTransition.class;
}

- (void)viewWillLayoutSubviews
{
    [super viewWillLayoutSubviews];
    
    CGFloat w = (self.view.frame.size.width - 10.0 * 4) / 3.0;
    UICollectionViewFlowLayout *layout = (UICollectionViewFlowLayout *)self.collectionViewLayout;
    layout.itemSize = CGSizeMake(w, w);
    layout.minimumLineSpacing = 9.0f;
    layout.minimumInteritemSpacing = 9.0f;
    layout.sectionInset = UIEdgeInsetsMake(10.0, 10.0, 10.0, 10.0);
}


#pragma mark <UICollectionViewDataSource>

- (NSInteger)numberOfSectionsInCollectionView:(UICollectionView *)collectionView {
    return 1;
}


- (NSInteger)collectionView:(UICollectionView *)collectionView numberOfItemsInSection:(NSInteger)section {
    return 100;
}

- (UICollectionViewCell *)collectionView:(UICollectionView *)collectionView cellForItemAtIndexPath:(NSIndexPath *)indexPath {
    WZAnimatedCollectionViewCell *cell = [collectionView dequeueReusableCellWithReuseIdentifier:reuseIdentifier forIndexPath:indexPath];
    
    return cell;
}

@end

File: Example/WZRootNavigationController/ViewControllers/WZCustomAnimatedTransitionListCollectionViewController.swift
//
//  WZCustomAnimatedTransitionListCollectionViewController.swift
//  WZRootNavigationController
//
//  Created by 吴哲 on 2018/4/24.
//  Copyright © 2018年 CocoaPods. All rights reserved.
//

import UIKit
import WZRootNavigationController
private let reuseIdentifier = "customAnimatedTransitionListCell"

class WZCustomAnimatedTransitionListCell: UICollectionViewCell {
    
    @IBOutlet weak var imageView: UIImageView!
    
    override init(frame: CGRect) {
        super.init(frame: frame)
    }
    
    required init?(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)
    }
    
}

class WZCustomAnimatedTransitionListCollectionViewController: UICollectionViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        self.navigationController?.hidesBarsOnSwipe = true
    }
    
    override var wz_animatedTransitionPluginClass: WZAnimatedTransitionPlugin.Type{
        return WZCustomAnimatedTransition.self
    }

    // MARK: UICollectionViewDataSource
    
    override func numberOfSections(in collectionView: UICollectionView) -> Int {
        return 1
    }
    
    
    override func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
        return 100
    }
    
    override func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: reuseIdentifier, for: indexPath) as! WZCustomAnimatedTransitionListCell
        cell.contentView.backgroundColor = .yellow
        
        return cell
    }
    
    override func viewWillLayoutSubviews() {
        super.viewWillLayoutSubviews()
        if let layout = self.collectionViewLayout as? UICollectionViewFlowLayout {
            let w:CGFloat = (self.view.frame.width - 10.0 * 4.0) / 3.0
            layout.itemSize = CGSize(width: w, height: w)
            layout.minimumLineSpacing = 9.0
            layout.minimumInteritemSpacing = 9.0
            layout.sectionInset = UIEdgeInsetsMake(10.0, 10.0, 10.0, 10.0)
        }
    }
}
